src/components/FeatureSelectDropdown/index.tsx

```diff
@@ -41,17 +37,17 @@ export default class AxisDropDown extends React.Component<AxisDropDownProps> {
         });
 
         const selectedOptionLabel = selectOptions
-            .find((option) => option.key === value)
+            .find((option) => option && "key" in option && option.key === value)
```

Comment:
Consider removing the forced assignment on line 30 (`const selectOptions: SelectProps["options"]`) which might let you remove the check for key?